MapboxDirections/Match/MBTracepoint.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+@objc(MBTracePoint)
+public class TracePoint: Waypoint {
+    
+    @objc open var alternateCount: Int
+    
+    @objc open var waypointIndex: Int
```

Comment:
Would it be more convenient for the Tracepoint object to maintain an index to a waypoint or a weak pointer to the Waypoint object itself?

Reply:
I think for 100% consistence sake, we should define this property with what the API is giving us.

Reply:
Strict consistency with the Directions API hasn’t ever been a high priority for this library. At least, consistency with the API has always been subordinate to intuitiveness. The API suffers from a lot of technical debt for backwards compatibility reasons that don’t apply to this library. Think about how the developer would need to use this library and whether there’s any boilerplate we can take off the developer’s hands.

In this particular case, the API uses indices because JSON doesn’t have a concept of [object references](https://en.wikipedia.org/wiki/JSON#Object_references). Just as this library already pairs steps with their origin and destination waypoints, I’m pretty sure a developer would expect us to pair tracepoints with waypoints, rather than make them `zip` the objects themselves.